test/message/console_low_stack_space.js

```diff
@@ -6,17 +6,26 @@ global.console = {};
 
 require('../common');
 
+// This test checks that, if Node cannot put together the `console` object
+// because it is low on stack space while doing so, it can succeed later
+// once the stack has unwound a little, and `console` is in a usable state then.
+
+let compiledConsole;
+
 function a() {
   try {
     return a();
   } catch (e) {
-    const console = consoleDescriptor.get();
-    if (console.log) {
-      console.log('Hello, World!');
+    compiledConsole = consoleDescriptor.get();
+    if (compiledConsole.log) {
+      // Using `console.log` itself might not succeed yet, but the code for it
+      // has been compiled.
     } else {
       throw e;
```

Comment:
IMHO you should throw a new Error, one that explicitly represents the failure to compile (could be prepared in advance if there's no room on the stack).